modules/scripts-utils/ace-adapter/src/main/resources/xml/xproc/library.xpl

```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<p:library xmlns:p="http://www.w3.org/ns/xproc" 
+    xmlns:px="http://www.daisy.org/ns/pipeline/xproc" 
+    xmlns:pxi="http://www.daisy.org/ns/pipeline/xproc/internal" version="1.0"
+    xmlns:c="http://www.w3.org/ns/xproc-step">
+
+
+    <p:declare-step type="px:ace" name="main">
+        <p:documentation xmlns="http://www.w3.org/1999/xhtml">
+            <h1 px:role="name">ACE report generation</h1>
+            <p px:role="desc">Generate an accessibility check report using the Accessibility Checker for EPUB tool.</p>
+        </p:documentation>
+        <!-- anyFileURI to the epub -->
+        <p:option name="epub" required="true">
+            <p:documentation xmlns="http://www.w3.org/1999/xhtml">
+                <h1 px:role="name">EPUB URI</h1>
+                <p px:role="desc">URI of the EPUB to control</p>
+            </p:documentation>
+        </p:option>
+        <!--  -->
+        <p:option name="temp-dir" required="false" select="''">
+            <p:documentation xmlns="http://www.w3.org/1999/xhtml">
+                <h1 px:role="name">Temporary directory URI</h1>
+                <p px:role="desc">Intermediate directory where the unzipped epub and html and json ACE reports will be stored</p>
+            </p:documentation>
+        </p:option>
+
+        <p:option name="output-dir" required="true">
```

Comment:
When we discussed how to return the HTML report, I thought the idea was to return a `c:result` with the result URI, like you did, but I didn't know you were going to add also a `output-dir` option. I guess it's fine, but there's not really a reason for not specifying the full output path instead of the directory I think, or is there? The option could also be dropped completely, and the report could be put inside the `temp-dir` (this is what I had in mind initially).

Reply:
Well my idea was to output the content of the html report on the result port of the exposed step, and the uri from the internal step.
I can directly output the URI of both report (html and json) from the exposed step within the temp directory, which could become the only optionnal argument.
I'll update that on the next commit.

Reply:
Ah yes I see, it's just an internal step. In that case I think it's indeed better to just drop output-dir.

Reply:
Saxon cannot load the ace html report directly (i have to use the html-load step for that in the exposed step to load and return the html) so the `c:result` was just a way to return the data (similarly as a `p:store` step like we discussed on slack).

I'm thinking on using a fileset to return both html and json URIs (see @rdeltour request) instead of the `c:result` in the internal step.

Reply:
Yes I know. But I was confused by the documentation of the result port. You are already returning the HTML content, but the documentation is wrong, it says it is a `c:result` document. This should be moved to the `result` port of the `pxi:ace` declaration.

I would use separate outputs for the HTML and the JSON. A fileset could be used for this in theory, the caller could filter it based on the media-type (html vs. json). But we normally use filesets for files that form a whole, like a DAISY or EPUB publication. This is a bit of a different use case I think, because we are interested in the individual files. What if, hypothetically, the files wouldn't be distinguishable based on media-type? What if the HTML or JSON or both aren't in the fileset, but you expect them to be?